### PR TITLE
mergeSpans requires all elements to be in the linelist

### PR DESCRIPTION
**Pull request by nigelmegitt**
Fix issue with empty spans or spans that return 0 width due to ligature with previous span

### CODE REVIEW

**Comment by nigelmegitt** [comment]
Since I've opened this on behalf of @JibberJim I can't approve it, but if I could, I would! It certainly seems to resolve the ligature problems that we saw. I checked for whitespace having an impact, but couldn't make it do so.

I note that there are test scripts in the repo, which compare the generated images against test renders, and we should probably check they are okay.

**Comment by palemieux** [comment]
> I note that there are test scripts in the repo, which compare the generated images against test renders, and we should probably check they are okay.

The renders can be generated using https://github.com/sandflow/imscJS/blob/master/src/test/webapp/gen-renders.html and the resulting PNGs can be compared using https://github.com/sandflow/imscJS/blob/master/script/refpngcompare.py (imagemagick is a dependency)

**Review by domjacks — APPROVED** [review]

**Pull request merged**
